unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/NFTShape/Handler/ECSNFTShapeComponentHandler.cs

```diff
@@ -0,0 +1,132 @@
+﻿using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using DCL.Helpers;
+using DCL.Helpers.NFT;
+using NFTShape_Internal;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class ECSNFTShapeComponentHandler : IECSComponentHandler<PBNFTShape>
+    {
+        internal INFTShapeFrameFactory factory;
+        
+        internal INFTInfoRetriever infoRetriever;
+        internal INFTAssetRetriever assetRetriever;
+        internal INFTShapeFrame shapeFrame;
+
+        private PBNFTShape model;
+        private DataStore_ECS7 dataStore;
+        
+        public ECSNFTShapeComponentHandler(INFTShapeFrameFactory factory, DataStore_ECS7 dataStoreEcs7, INFTInfoRetriever infoRetriever, INFTAssetRetriever assetRetriever)
+        {
+            this.factory = factory;
+            dataStore = dataStoreEcs7;
+
+            this.infoRetriever = infoRetriever;
+            this.assetRetriever = assetRetriever;
+        }
+        
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            infoRetriever.Dispose();
+            assetRetriever.Dispose();
+            DisposeShapeFrame(entity);
+        }
+
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBNFTShape model)
+        {
+            // We create the frame gameobject
+            if(shapeFrame == null || this.model.Style != model.Style)
+                CreateNFTShapeFrame(entity, model);
+            
+            // We apply the model to the frame
+            ApplyModel(model);
+
+            // We load the NFT image
+            LoadNFT(scene,model);
+        }
+
+        private void DisposeShapeFrame(IDCLEntity entity)
+        {
+            if (shapeFrame == null)
+                return;
+            
+            shapeFrame.Dispose();
+            ECSComponentsUtils.DisposeMeshInfo(entity.meshesInfo);
+            GameObject.Destroy(shapeFrame.gameObject);
+        }
+        
+        internal async void LoadNFT(IParcelScene scene,PBNFTShape model)
+        {
+            dataStore.AddPendingResource(scene.sceneData.id, model);
```

Comment:
we are not currently expecting for the nft image to be downloaded during the loading, since the api is sort of unreliable.
I think we shouldn't be adding it as a pending resource